bin/jruby.bash

```diff
@@ -140,17 +140,17 @@ else
   fi
 fi
 
-# Detect Java 9+ by the presence of a jmods directory in JAVA_HOME
-if [ -d "$JAVA_HOME"/jmods ]; then
-  is_java9=1
+# Detect modularized Java
+if [ -f ${JAVA_HOME}/lib/modules ] || [ -f ${JAVA_HOME}/release ] && grep -q ^MODULES ${JAVA_HOME}/release; then
```

Comment:
Hi @headius, was the `"` drop around `$JAVA_HOME` intentional here? Asking, because I am seeing some errors in 9.2.17.0 which seems to be related to me having spaces in my `$JAVA_HOME`:

```
$ ruby --version
/Users/robin/.rubies/jruby-9.2.17.0/bin/ruby: line 144: [: /Library/Internet: binary operator expected
/Users/robin/.rubies/jruby-9.2.17.0/bin/ruby: line 144: [: /Library/Internet: binary operator expected
jruby 9.2.17.0 (2.5.8) 2021-03-29 84d363da97 Java HotSpot(TM) 64-Bit Server VM 25.281-b09 on 1.8.0_281-b09 +jit [darwin-x86_64]

$ unset JAVA_HOME

$ ruby --version 
jruby 9.2.17.0 (2.5.8) 2021-03-29 84d363da97 OpenJDK 64-Bit Server VM 15.0.1+9-18 on 15.0.1+9-18 +jit [darwin-x86_64]
```

Reply:
It was not intentional and I even thought at one point I need to restore those quotes. You can make the modification yourself locally, or use the native launcher from jruby-launcher gem which should not have this issue. I will open a PR to fix this in JRuby proper.

Reply:
FWIW we do recommend not having the JDK installed in a path with spaces for many other reasons, including the complexity of making sure re-launched Java/JRuby subprocesses use the proper quoting. Thank you for bringing this issue to my attention again, I have pushed https://github.com/jruby/jruby/pull/6638 to fix it.

Reply:
@walro Could you share more information about your system in the PR? For me, JDKs are installed in `/Library/Java/JavaVirtualMachines` so this `/Library/Internet Something` path may be a new thing we have to be prepared for.

Reply:
Absolutely, done so!